src/librekey/key_store_g10.c

```diff
@@ -441,6 +475,308 @@ parse_seckey(pgp_keydata_key_t *keydata, s_exp_t *s_exp, pgp_pubkey_alg_t alg)
     return true;
 }
 
+static pgp_seckey_t *
+g10_decrypt_seckey(const pgp_key_t *key, FILE *passfp)
+{
+    uint8_t        derived_key[PGP_MAX_KEY_SIZE];
+    char *         passphrase;
+    char           pass[MAX_PASSPHRASE_LENGTH];
+    unsigned       keysize;
+    uint8_t *      decrypted;
+    pgp_seckey_t * seckey;
+    s_exp_t        s_exp = {};
+    size_t         output_written = 0;
+    size_t         input_consumed = 0;
+    botan_cipher_t decrypt;
+
+    if (key->key.seckey.encrypted_len <= 0) {
+        fprintf(stderr, "Hasn't got encrypted data!\n");
+        return NULL;
+    }
+
+    if (pgp_getpassphrase(passfp, pass, sizeof(pass)) == 0) {
+        pass[0] = '\0';
+    }
```

Comment:
It does return false when fgets fails (when using passfp and EOF is hit, for example) so it's better to fail here. Also since `pgp_getpassphrase` returns a bool, you could probably just do something like `if (!pgp_getpassphrase(...)) { return false; }`.